### PR TITLE
K-hop neighbors

### PR DESCRIPTION
**Pull request by seunghwak**
K-hop neighbors implementation (mainly to support 2-hop neighbors).

### CODE REVIEW

**Comment by seunghwak** [comment]
rerun tests

File: cpp/tests/traversal/k_hop_nbrs_test.cpp [hunk 1]
```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2020-2022, NVIDIA CORPORATION.
```
Comment (ChuckHastings) [inline cpp/tests/traversal/k_hop_nbrs_test.cpp:2]: Probably just 2022

**Review by ChuckHastings — APPROVED** [review]
Looks good to me.  One little copyright observation.

**Review by naimnv — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**